pkg/api/customization/cluster/actions_cis.go

```diff
@@ -73,16 +73,51 @@ func (a ActionHandler) runCisScan(actionName string, action *types.Action, apiCo
 			fmt.Sprintf("CIS scan already running on cluster"))
 	}
 
-	if cisScanConfig.OverrideBenchmarkVersion != "" {
-		_, err := a.CisBenchmarkVersionLister.Get(namespace.GlobalNamespace, cisScanConfig.OverrideBenchmarkVersion)
-		if err != nil {
-			if kerrors.IsNotFound(err) {
+	validBenchmarkVersions, err := a.getValidBenchmarkVersions()
+	if err != nil {
+		logrus.Errorf("error fetching benchmark versions: %v", err)
+		return httperror.WrapAPIError(err, httperror.ServerError, "error fetching cis benchmark versions %v")
+	}
+
+	if cisScanConfig.OverrideBenchmarkVersion != "" &&
+		!validBenchmarkVersions[cisScanConfig.OverrideBenchmarkVersion] {
+		return httperror.WrapAPIError(err, httperror.InvalidOption,
```

Comment:
wouldn't err be nil?

Reply:
copy, paste error, thanks for the catch.